Bug: Everflow's `IP_PROTOCOL` ACL rule can match LACP packets

### Is it platform specific

broadcom

### Importance or Severity

High

### Description of the bug

When running the `test_everflow_fwd_recircle_port_queue_check` sonic-mgmt test we see it sporadically fail on certain testbeds.
The issue is the test injects 50 packets but sees >50 packets on the Ethernet-Rec0 counter:
```
E                   AssertionError: Recircle port Ethernet-Rec0 queue5 counter value is not same as packets sent
(Pdb) p expected_pkts
50
```
```
root@ctn101:~# show queue counters Ethernet-Rec0
Last cached time was 2026-04-13T16:26:35.640947
Ethernet-Rec0 Last cached time was 2026-04-13T16:26:35.640947
         Port    TxQ    Counter/pkts    Counter/bytes    Drop/pkts    Drop/bytes
-------------  -----  --------------  ---------------  -----------  ------------
Ethernet-Rec0    UC0               0                0            0             0
Ethernet-Rec0    UC1               0                0            0             0
Ethernet-Rec0    UC2               0                0            0             0
Ethernet-Rec0    UC3               0                0            0             0
Ethernet-Rec0    UC4               0                0            0             0
Ethernet-Rec0    UC5              52            8,004            0             0
Ethernet-Rec0    UC6               0                0            0             0
Ethernet-Rec0    UC7               0                0            0             0
```

The problem is RULE_7 is being hit by LACP packets randomly:
```
root@ctn101:~# aclshow -a
RULE NAME    TABLE NAME      PRIO    PACKETS COUNT    BYTES COUNT
-----------  ------------  ------  ---------------  -------------
RULE_1       EVERFLOW        9999              120          12480
<snip>
RULE_7       EVERFLOW        9993                4            512
```

RULE_7 being the IP_PROTOCOL rule:
```
root@ctn101:~# show acl rule
Table     Rule       Priority  Action                          Match                         Status
--------  -------  ----------  ------------------------------  ----------------------------  --------
EVERFLOW  RULE_1         9999  MIRROR INGRESS: test_session_1  SRC_IP: 20.0.0.10/32          Active
<snip>
EVERFLOW  RULE_7         9993  MIRROR INGRESS: test_session_1  IP_PROTOCOL: 126              Active
```

I captured the sporadic packet hitting RULE_7 and we can see it's a LACP packet:
```
0180 c200 0002 622d 5171 ed0b **8809**
0101 0114 8000 1655 9c**7e** 3194 0001 8000
0001 3d00 0000 0214 ffff 9c69 edce fccd
046c 00ff 0079 3d00 0000 0310 0000 0000
0000 0000 0000 0000 0000 0000 0000 0000
0000 0000 0000 0000 0000 0000 0000 0000
0000 0000 0000 0000 0000 0000 0000 0000
0000 0000 0000 0000 0000 0000 0000
```

We can also see that 10-bytes after the Ethernet header (same offset as IP_PROTOCOL into an IPv4 header) we see a `0x7e` which is `126` in decimal which matches our rule `IP_PROTOCOL: 126`.

This section of the LACP header is the Actor System ID (16:55:9c:**7e**:31:94) so it can hit depending on the LACP's peer mac.

I brought this up with BRCM in CS00012457180 and they mentioned that the ACL rule is underqualified.
We need the IP_PROTOCOL rule to also have IP_TYPE included in it to avoid matching on non-IP packets.

### Steps to Reproduce

Run `test_everflow_fwd_recircle_port_queue_check` on a testbed that has a peer device with `7e` as the 4th byte of the system MAC address.

### Actual Behavior and Expected Behavior

The `IP_PROTOCOL` ACL is matching against LACP packets when it should only match against IP packets.

### Relevant log output

```shell

```

### Output of `show version`

```shell

```

### Attach files (if any)

_No response_

--- a/acl_loader/main.py
+++ b/acl_loader/main.py
@@ -600,6 +600,11 @@ class AclLoader(object):
                         rule.ip.config.protocol, table_name, rule_idx))
 
                 rule_props["IP_PROTOCOL"] = rule.ip.config.protocol
+            if "IP_PROTOCOL" in rule_props:
+                # If we don't include IP_TYPE as a qualifier in the IP_PROTOCOL rule
+                # we could match on non-IP packets if the bits at the same offset match
+                # https://github.com/sonic-net/sonic-mgmt/issues/23960
+                rule_props["IP_TYPE"] = "IP"
 
         if rule.ip.config.source_ip_address:
             source_ip_address = rule.ip.config.source_ip_address

--- a/acl_loader/main.py
+++ b/acl_loader/main.py
@@ -600,11 +600,6 @@ class AclLoader(object):
                         rule.ip.config.protocol, table_name, rule_idx))
 
                 rule_props["IP_PROTOCOL"] = rule.ip.config.protocol
-            if "IP_PROTOCOL" in rule_props and "IP_TYPE" not in rule_props:
-                # If we don't include IP_TYPE as a qualifier in the IP_PROTOCOL rule
-                # we could match on non-IP packets if the bits at the same offset match
-                # https://github.com/sonic-net/sonic-mgmt/issues/23960
-                rule_props["IP_TYPE"] = "IP"
 
         if rule.ip.config.source_ip_address:
             source_ip_address = rule.ip.config.source_ip_address
@@ -778,6 +773,12 @@ class AclLoader(object):
         deep_update(rule_props, self.convert_icmp(table_name, rule_idx, rule))
         deep_update(rule_props, self.convert_transport(table_name, rule_idx, rule))
         deep_update(rule_props, self.convert_input_interface(table_name, rule_idx, rule))
+
+        if "IP_PROTOCOL" in rule_props and "IP_TYPE" not in rule_props:
+            # If we don't include IP_TYPE as a qualifier in the IP_PROTOCOL rule
+            # we could match on non-IP packets if the bits at the same offset match
+            # https://github.com/sonic-net/sonic-mgmt/issues/23960
+            rule_props["IP_TYPE"] = "IP"
 
         self.validate_rule_fields(rule_props)
 

--- a/acl_loader/main.py
+++ b/acl_loader/main.py
@@ -600,7 +600,7 @@ class AclLoader(object):
                         rule.ip.config.protocol, table_name, rule_idx))
 
                 rule_props["IP_PROTOCOL"] = rule.ip.config.protocol
-            if "IP_PROTOCOL" in rule_props:
+            if "IP_PROTOCOL" in rule_props and "IP_TYPE" not in rule_props:
                 # If we don't include IP_TYPE as a qualifier in the IP_PROTOCOL rule
                 # we could match on non-IP packets if the bits at the same offset match
                 # https://github.com/sonic-net/sonic-mgmt/issues/23960

--- a/tests/acl_loader_test.py
+++ b/tests/acl_loader_test.py
@@ -76,6 +76,7 @@ class TestAclLoader(object):
                 "VLAN_ID": 369,
                 "ETHER_TYPE": "2048",
                 "IP_PROTOCOL": 6,
+                'IP_TYPE': 'IP',
                 "SRC_IP": "20.0.0.2/32",
                 "DST_IP": "30.0.0.3/32",
                 "PACKET_ACTION": "FORWARD",
@@ -94,6 +95,7 @@ class TestAclLoader(object):
             "VLAN_ID": 369,
             "ETHER_TYPE": "2048",
             "IP_PROTOCOL": 6,
+            'IP_TYPE': 'IP',
             "SRC_IP": "20.0.0.2/32",
             "DST_IP": "30.0.0.3/32",
             "PACKET_ACTION": "FORWARD",
@@ -134,6 +136,7 @@ class TestAclLoader(object):
             "VLAN_ID": 369,
             "ETHER_TYPE": 2048,
             "IP_PROTOCOL": 6,
+            'IP_TYPE': 'IP',
             "SRC_IP": "20.0.0.2/32",
             "DST_IP": "30.0.0.3/32",
             "PACKET_ACTION": "FORWARD",
@@ -147,6 +150,7 @@ class TestAclLoader(object):
         assert acl_loader.rules_info[("DATAACLV4V6", "RULE_2")] == {
             "ETHER_TYPE": 34525,
             "IP_PROTOCOL": 58,
+            'IP_TYPE': 'IP',
             "SRC_IPV6": "::1/128",
             "DST_IPV6": "::1/128",
             "PACKET_ACTION": "FORWARD",
@@ -170,6 +174,7 @@ class TestAclLoader(object):
             "ICMP_TYPE": 3,
             "ICMP_CODE": 0,
             "IP_PROTOCOL": 1,
+            'IP_TYPE': 'IP',
             "SRC_IP": "20.0.0.2/32",
             "DST_IP": "30.0.0.3/32",
             "ETHER_TYPE": "2048",
@@ -208,6 +213,7 @@ class TestAclLoader(object):
             "ICMP_TYPE": 136,
             "ICMP_CODE": 0,
             "IP_PROTOCOL": 1,
+            'IP_TYPE': 'IP',
             "PACKET_ACTION": "FORWARD",
             "PRIORITY": "9998"
         }
@@ -220,6 +226,7 @@ class TestAclLoader(object):
             "ICMPV6_TYPE": 136,
             "ICMPV6_CODE": 0,
             "IP_PROTOCOL": 58,
+            'IP_TYPE': 'IP',
             "PACKET_ACTION": "FORWARD",
             "PRIORITY": "9998"
         }

